consultation_analyser/consultations/models.py

```diff
@@ -40,301 +41,237 @@ class Meta:
+        ]
 
 
-class Section(UUIDPrimaryKeyModel, TimeStampedModel):
+class Question(UUIDPrimaryKeyModel, TimeStampedModel, SlugFromTextModel):
+    text = models.TextField()
     consultation = models.ForeignKey(Consultation, on_delete=models.CASCADE)
-    name = models.TextField()
-    slug = models.CharField(null=False, max_length=256)
+    order = models.IntegerField(null=True)
 
-    class Meta:
+    class Meta(UUIDPrimaryKeyModel.Meta, TimeStampedModel.Meta, SlugFromTextModel.Meta):
         constraints = [
-            models.UniqueConstraint(
-                fields=["slug", "consultation"], name="unique_section_consultation"
-            ),
+            models.UniqueConstraint(fields=["slug"], name="unique_q_slug"),
         ]
 
 
-class MultipleChoiceNotProportionalError(Exception):
-    pass
-
-
-@dataclass
-class MultipleChoiceQuestionStats:
-    question: str
-    counts: dict[str, int]
-    total_responses: int
-    has_multiple_selections: bool
+class QuestionPart(UUIDPrimaryKeyModel, TimeStampedModel):
+    class QuestionType(models.TextChoices):
+        FREE_TEXT = "free_text"
+        SINGLE_OPTION = "single_option"
+        MULTIPLE_OPTIONS = "multiple_options"
 
-    def percentages(self):
-        if self.has_multiple_selections:
-            raise MultipleChoiceNotProportionalError(
-                "It does not make sense to calculate percentages for a multiple choice question supporting multiple selections"
-            )
-        else:
-            return {
-                option: round((float(count) / self.total_responses) * 100)
-                for option, count in self.counts.items()
-            }
-
-
-class Question(UUIDPrimaryKeyModel, TimeStampedModel):
-    MULTIPLE_CHOICE_COUNTS_QUERY = """
-        SELECT question_id as id, elem ->> 'question_text' as question, option, COUNT(*) as count
-        FROM    consultations_answer,
-                jsonb_array_elements(multiple_choice) AS elem,
-                jsonb_array_elements_text(elem -> 'options') AS option
-        WHERE question_id = %s
-        GROUP BY question_id, option, question
-        ORDER BY question, option ASC
-    """
-
-    # until we explicitly annotate these questions, use the presence
-    # of multiple selections to infer whether they're supported
-    HAS_MULTIPLE_SELECTIONS_QUERY = """
-        SELECT EXISTS (
-            SELECT 1
-            FROM consultations_answer,
-                 jsonb_array_elements(multiple_choice) AS elem
-            WHERE question_id = %s
-            AND elem ->> 'question_text' = %s
-            AND jsonb_array_length(elem -> 'options') > 1
-        ) AS has_multiple_selections;
-    """
+    question = models.ForeignKey(Question, on_delete=models.CASCADE)
+    text = models.TextField()
+    type = models.CharField(max_length=16, choices=QuestionType.choices)
+    options = models.JSONField(default=list)
+    order = models.IntegerField(null=True)
```

Comment:
Same questions as for `class Question`

Reply:
And same response - what you said above seems sensible.

Reply:
Changed as described above for `Question`.